avionics/common/src/radio.cpp

```diff
@@ -129,109 +135,89 @@ void Radio::send() {
 
 void Radio::sendStatus(uint32_t time, RocketStatus status,
                        SensorCollection &sensors, IgnitorCollection &ignitors) {
-    PacketBuffWriter buf;
-    addIdTime(buf, Ids::status_ping, time);
-
-    buf.write(&status, 1);
+    self.tx_q_.allocSubpkt(10);
```

Comment:
I'm still a little concerned about this sum being a point of failure. I think we've had issues with that in the past.

`tx_q_.write` asserts if we write more than allocated right? But we still would have issues if we write less.

What if allocSubpkt returns a new PacketBuffWriter that writes directly into the buffer. But on destruction PacketBuffWriter asserts that the number of bytes is exactly equal to what was allocated? Just an idea. I'm open to other ideas as well, I just think that it would be nice if we didn't have to make any sacrifices for these changes.

Reply:
`tx_q_` also does a check if you write too little (on the next time you call allocSubpkt, since that's really the only time it can check); it also pulls up `pos_` to the right position anyways to be sure: 
```
    // previous usage of write() should've brought pos_ to the right place,
    // since otherwise the subpacket wouldn't be filled
    assert(pos_ == deref(subpkts_end));

    // but it's safest/more robust to enforce it anyways
    // Reset data pointer to newest subpacket (for use in write())
    pos_ = deref(subpkts_end);
```

Reply:
Ah okay cool. That works.